Controller/Component/WizardComponent.php

```diff
@@ -510,6 +529,9 @@ protected function _validStep($step) {
  * @return void
  */
 	protected function _setCurrentStep($step) {
+		if (!in_array($step, $this->steps)) {
+			return;
+		}
```

Comment:
I do not remember exactly why that was needed. I think that has something to do with internal workings of this component to prevent setting the future step that is not in $this->steps yet. An infinite loop happened without this check. Notice `while (current($this->steps) != $step)` below. The clients cannot call _setCurrentStep() directly as it is protected so there is no need to worry.